src/DomainReader.ts

```diff
@@ -108,6 +108,58 @@ export class DomainReader {
     throw Error(ERROR_MESSAGES.RESOLVER_NOT_SUPPORTED)
   }
 
+  public getAllRolesHistory = async (): Promise<IRoleDefinition[]> => {
```

Comment:
Maybe we should think about the return type. Maybe we want a map of arrays? Where the key is a role name and the value is all of the `IRoleDefinition` over time?